### PR TITLE
Add higher-res icons for iOS, Android and Windows 10

### PR DESCRIPTION
**Pull request by Kussie**
Re did the favicon and added support for iOS, Android and Windows 10 higher res icons (Useful for bookmarking or saving shortcut on device homescreen)

### CODE REVIEW

**Comment by gotson** [comment]
Can you test this with a servlet context path? There are specific transformations to handle this and we need to test that.

File: komga-webui/public/index.html [hunk 1]
```diff
@@ -6,6 +6,11 @@
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <title>Komga</title>
     <link rel="icon" href="/favicon.ico" th:href="@{/favicon.ico}">
+    <link rel="apple-touch-icon" sizes="180x180" th:href="@{/apple-touch-icon.png}">
+    <link rel="icon" type="image/png" sizes="32x32" th:href="@{/favicon-32x32.png}">
+    <link rel="icon" type="image/png" sizes="16x16" th:href="@{/favicon-16x16.png}">
```
Comment (gotson) [inline komga-webui/public/index.html:11]: Those should have a href like the existing favicon

Comment (Kussie) [reply komga-webui/public/index.html]: Fixed, completely missed that

File: komga/src/main/kotlin/org/gotson/komga/infrastructure/web/WebMvcConfiguration.kt [hunk 2]
```diff
@@ -30,11 +30,21 @@ class WebMvcConfiguration : WebMvcConfigurer {
...
       )
       .addResourceLocations(
         "classpath:public/index.html",
-        "classpath:public/favicon.ico"
+        "classpath:public/favicon.ico",
+        "classpath:public/favicon-16x16.png",
+        "classpath:public/favicon-32x32.png",
+        "classpath:public/mstile-144x144.png",
+        "classpath:public/apple-touch-icon.png",
+        "classpath:public/apple-touch-icon-180x180.png"
```
[the remark this answers is not in the document]
Comment (Kussie) [reply komga/src/main/kotlin/org/gotson/komga/infrastructure/web/WebMvcConfiguration.kt]: opps forgot to remove those.  All cleaned up

**Comment by Kussie** [comment]
Should be all good now :)

File: komga-webui/public/index.html [hunk 3]
```diff
@@ -6,6 +6,11 @@
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <title>Komga</title>
     <link rel="icon" href="/favicon.ico" th:href="@{/favicon.ico}">
+    <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png" th:href="@{/apple-touch-icon.png}">
+    <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png" th:href="@{/favicon-32x32.png}">
+    <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png" th:href="@{/favicon-16x16.png}">
+    <meta name="msapplication-TileColor" content="#08397f">
+    <meta name="msapplication-TileImage" content="/mstile-144x144.png">
```
Comment (gotson) [inline komga-webui/public/index.html:13]: I'm surprised this works with a servlet context path though 🤔

**Comment by gotson** [comment]
You probably need something here too https://github.com/gotson/komga/blob/a6c47f1b85d11eca54680ea4370d3660521d1e3c/komga-webui/vue.config.js#L14

**Comment by Kussie** [comment]
> You probably need something here too
> 
> https://github.com/gotson/komga/blob/a6c47f1b85d11eca54680ea4370d3660521d1e3c/komga-webui/vue.config.js#L14

Good call, missed that

**Comment by gotson** [comment]
> > You probably need something here too
> > https://github.com/gotson/komga/blob/a6c47f1b85d11eca54680ea4370d3660521d1e3c/komga-webui/vue.config.js#L14
> 
> Good call, missed that

It's kinda tricky how that all works 😅

**Comment by gotson** [comment]
> Re did the favicon

What changed exactly?

**Comment by Kussie** [comment]
> > Re did the favicon
> 
> What changed exactly?

I took the nice high quality SVG and ran it through a little generator i have to generate most of the current favicon solutions.  Though pretty sure the favicon.ico might have been entirely valid somehow.  As i never used to see the preview of it in the file explorer (Despite having an extension that shows ico files), but since regenerating it shows up perfectly, so included it as well.

File: komga-webui/public/index.html (hunk 3 above)
Comment (Kussie) [reply komga-webui/public/index.html]: Fixed

**Pull request merged**